crates/db_schema/src/impls/multi_community.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{
+  newtypes::{CommunityId, MultiCommunityId, PersonId},
+  source::multi_community::{MultiCommunity, MultiCommunityInsertForm},
+  utils::{get_conn, DbPool},
+};
+use diesel::{
+  dsl::{delete, insert_into},
+  result::Error,
+  ExpressionMethods,
+  QueryDsl,
+};
+use diesel_async::{scoped_futures::ScopedFutureExt, AsyncConnection, RunQueryDsl};
+use lemmy_db_schema_file::schema::{multi_community, multi_community_entry};
+
+impl MultiCommunity {
+  pub async fn create(
+    pool: &mut DbPool<'_>,
+    form: &MultiCommunityInsertForm,
+  ) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    insert_into(multi_community::table)
+      .values(form)
+      .get_result::<Self>(conn)
+      .await
+  }
+
+  pub async fn upsert(
```

Comment:
You probably won't need this after switching to atomic updates.

Reply:
This is needed for federation which uses a full list of communities, so it cant be updated one by one.